tests/b2share_functional_tests/test_record_cli.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of EUDAT B2Share.
+# Copyright (C) 2017 SurfSara, University of Tübingen
+#
+# B2Share is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# B2Share is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with B2Share; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Test B2Share records cli module."""
+from __future__ import absolute_import, print_function
+
+from click.testing import CliRunner
+from flask_cli import ScriptInfo
+
+import b2share.modules.records.cli as records_cli
+#; import check_dois
+
+headers = [('Content-Type', 'application/json'),
+           ('Accept', 'application/json')]
+patch_headers = [('Content-Type', 'application/json-patch+json'),
+                 ('Accept', 'application/json')]
+
+def test_record_doi_cli(app, test_community, test_records, login_user):
+    """Test adding a community and setting its schema using CLI commands."""
+    with app.app_context():
+        record_id_list = []
+        for record in test_records:
+            with app.test_client() as client:
+                login_user(test_community.admin, client)
+                record_id_list.append(record.pid)
+
+        runner = CliRunner()
+        script_info = ScriptInfo(create_app=lambda info: app)
+
+        # hijacking internal functions
+        ref_called = 0
+        register_called = 0
+
+        def _my_ref(doi_value):
+            nonlocal ref_called
+            ref_called += 1
+
+        def _my_register(doi, url, doc):
+            nonlocal register_called
+            register_called += 1
+
+        records_cli._datacite_doi_reference = _my_ref
```

Comment:
This can be done in a better way with `unittest.mock`.
```python
with mock.patch.multiple('records_cli', reference='_datacite_doi_reference', register='_datacite_register_doi') as mocks:
    _datacite_register_doi = mocks['register']
    _datacite_doi_reference = mocks['reference']
    _datacite_doi_reference = ...
    ...
    assert datacite_doi_reference.assert_called_with(<doi_value>)
    assert datacite_doi_reference.assert_called_with(<doi, url, doc>)
```
This enable to replace the functions only for this test even if the test fails and raise an exception.